model/riscv_validate_config.sail

```diff
@@ -0,0 +1,117 @@
+/*=======================================================================================*/
+/*  This Sail RISC-V architecture model, comprising all files and                        */
+/*  directories except where otherwise noted is subject the BSD                          */
+/*  two-clause license in the LICENSE file.                                              */
+/*                                                                                       */
+/*  SPDX-License-Identifier: BSD-2-Clause                                                */
+/*=======================================================================================*/
+
+function check_privs() -> bool = {
+  if hartSupports(Ext_S) & not(hartSupports(Ext_U))
+  then {
+    print_endline("User mode (U) should be enabled if supervisor mode (S) is enabled.");
+    return false;
+  };
+  true
+}
+
+// This logic will need to change once MXL/SXL/UXL can differ.
+function check_mmu_config() -> bool = {
+  var valid : bool = true;
+  if xlen == 64 then {
+    if not(hartSupports(Ext_S)) & (hartSupports(Ext_Sv57) | hartSupports(Ext_Sv48) | hartSupports(Ext_Sv39))
+    then {
+      valid = false;
+      print_endline("Supervisor mode (S) disabled but one of (Sv57, Sv48, Sv39) is enabled: cannot support address translation without supervisor mode.");
+    };
+    if hartSupports(Ext_Sv57) & not(hartSupports(Ext_Sv48))
+    then {
+      valid = false;
+      print_endline("Sv57 is enabled but Sv48 is disabled: supporting Sv57 requires supporting Sv48.");
+    };
+    if hartSupports(Ext_Sv48) & not(hartSupports(Ext_Sv39))
+    then {
+      valid = false;
+      print_endline("Sv48 is enabled but Sv39 is disabled: supporting Sv48 requires supporting Sv39.");
+    };
+    if hartSupports(Ext_Sv32)
+    then {
+      valid = false;
+      print_endline("Sv32 is enabled: Sv32 is not supported on RV64.");
+    };
+  } else {
+    assert(xlen == 32);
+    if not(hartSupports(Ext_S)) & hartSupports(Ext_Sv32)
+    then {
+      valid = false;
+      print_endline("Supervisor mode (S) is disabled but Sv32 is enabled: cannot support address translation without supervisor mode.");
+    };
+    if hartSupports(Ext_Sv39)
+    then {
+      valid = false;
+      print_endline("Sv39 is enabled: Sv39 is not supported on RV32.");
+    };
+    if hartSupports(Ext_Sv48)
+    then {
+      valid = false;
+      print_endline("Sv48 is enabled: Sv48 is not supported on RV32.");
+    };
+    if hartSupports(Ext_Sv57)
+    then {
+      valid = false;
+      print_endline("Sv57 is enabled: Sv57 is not supported on RV32.");
+    };
+  };
+  valid
+}
+
+function check_vlen_elen() -> bool = {
+  // Should this be conditioned on a vector extension being enabled?
```

Comment:
I think it's simpler just to require a valid config anyway. We can make the defaults valid so it's no effort.